TopPlaces/TopPlaces/Controllers/DetailsPhotoViewController.m

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+#import "DetailsPhotoViewController.h"
+
+#import "PhotosHistory.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface DetailsPhotoViewController() <UIScrollViewDelegate>
+
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *spinner;
+@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
+@property (weak, nonatomic) IBOutlet UILabel *noImageLoadedView;
+@property (strong, nonatomic) UIImageView *imageView;
+@property (strong, nonatomic) NSURLSessionTask *downloadTask;
+
+@end
+
+@implementation DetailsPhotoViewController
+
+#pragma mark -
+#pragma mark UIViewController overrides
+#pragma mark -
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  self.scrollView.delegate = self;
+  UITapGestureRecognizer *doubleTapRecognizer =
+      [[UITapGestureRecognizer alloc] initWithTarget:self
+                                              action:@selector(scrollViewDoubleFingerTapped:)];
+  doubleTapRecognizer.numberOfTapsRequired = 2;
+  doubleTapRecognizer.numberOfTouchesRequired = 1;
+  [self.scrollView addGestureRecognizer:doubleTapRecognizer];
+  [self loadImage];
+  NSLog(@"DetailsPhotoViewController::viewDidLoad self %p", self);
+}
+
+
+- (void)scrollViewDoubleFingerTapped:(UITapGestureRecognizer*)recognizer {
+  //if already zoomed out - zoom in a bit
+  if (self.scrollView.zoomScale == self.scrollView.minimumZoomScale) {
+    [self.scrollView setZoomScale:self.scrollView.zoomScale*2 animated:YES];
+  } else {
+    //zoom out
+    [self.scrollView setZoomScale:self.scrollView.minimumZoomScale animated:YES];
+  }
+}
+
+
+- (void)viewDidAppear:(BOOL)animated {
+  NSLog(@"DetailsPhotoViewController::viewDidAppear");
+  [super viewDidAppear:animated];
+}
+
+
+- (void)viewDidLayoutSubviews {
+  NSLog(@"DetailsPhotoViewController::viewDidLayoutSubviews");
+  [super viewDidLayoutSubviews];
+
+  if (self.imageView) {
+    [self aspectFitImage];
+  }
+}
+
+- (void) viewDidDisappear:(BOOL)animated {
+  [super viewDidDisappear:animated];
+  NSLog(@"DetailsPhotoViewController::viewDidDisappear");
+  [self.downloadTask cancel];
```

Comment:
I think cancelling the task here might be problematic. This method gets called also when the user sends the app to the background. If it happens while a task is active when the user returns he will not see the image.
Instead consider cancelling on <code>viewDidUnload</code>.